Automattic/jetpack
# Writing Prompt block: use site language setting for translations

## Proposed changes:

Updates the Writing Prompt block:

- Uses the site language setting to fetch the prompt--presumable since the block is post content (rather than admin interface), it should use the site language setting, rather that the user language setting.
- Adds all relevant text strings to the blogging-prompts API response, so that we can translate them using the site language setting.
- Uses block attributes for storing text strings, so they can be stored in the site language by the save function.
- Only fetch the prompt one time, when the block is inserted, so that the block doesn't spontaneously update when opened in the editor.
- Use a background image for the svg icon in the block, to prevent block invalidation issues on multisite (when saving svgs to the post_content is not allowed).

### Other information:


## Jetpack product discussion

PT: pe7F0s-uo-p2
Related discussion: pxLjZ-7H2-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Sandbox public-api.wordpress.com
* Apply the changes from this PR to your sandbox (`bin/jetpack-downloader test jetpack update/writing-prompt-block-site-locale`)
* Make sure your Jetpack is connected to WP.com and beta blocks are enabled (`define( 'JETPACK_BETA_BLOCKS', true );`)
* Insert a writing prompts block and save the post.
* Reload the editor and verify there are no validation errors for the block.
* Change the site language setting in Settings > General to one of the mag16 locales.
* Create another post and insert a writing prompt block. See that all text in the block uses the language you set for the site.
* Save the post and make sure it displays correctly on the front-end.
* Load the first block you saved, and see that the language does not change--the prompt content is saved into the attributes of the block.
